proof/refine/ARM/Schedule_R.thy

```diff
@@ -1569,18 +1580,27 @@ lemma ksReadyQueuesL1Bitmap_st_tcb_at':
    apply simp
    done
 
+lemma curDomain_or_return_0:
+  "\<lbrakk> \<lbrace>P\<rbrace> curDomain \<lbrace>\<lambda>rv s. Q rv s \<rbrace>; \<And>s. P s \<Longrightarrow> ksCurDomain s \<le> maxDomain \<rbrakk>
+   \<Longrightarrow> \<lbrace>P\<rbrace> if 1 < numDomains then curDomain else return 0 \<lbrace>\<lambda>rv s. Q rv s \<rbrace>"
+  apply (case_tac "1 < numDomains"; simp)
+  apply (simp add: valid_def curDomain_def simpler_gets_def return_def maxDomain_def)
+  done
+
+lemma invs_no_cicd_ksCurDomain_maxDomain':
+  "invs_no_cicd' s \<Longrightarrow> ksCurDomain s \<le> maxDomain"
```

Comment:
nitpick: move this to Invariants_H and make it `elim!` like the similar rules (same for other architectures)

Reply:
Can you elaborate which similar rules you are thinking of? I'm not seeing any `invs_no_cicd'` derivations in Invariants_H. Happy to put it there, but would like more clarity of the idea. `invs_no_cicd'_queues` also lives in Schedule_R and probably was the same low level of inspiration.

Reply:
I was considering it as similar to rules like `invs_valid_objs'`. There's no big problem with leaving it as is, especially if `invs_no_cicd'_queues` is already there as well, but everything else being equal I do like having this sort of lemma early in the dependencies.